packages/aws-cdk-lib/aws-route53/lib/health-check.ts

```diff
@@ -0,0 +1,658 @@
+  if (props.routingControl === undefined) {
+    throw new Error(`RoutingControl is required for health check type: ${props.type}`);
+  }
+}
+
+function rulePortIsNotAllowed(props: HealthCheckProps) {
+  if (props.port) {
+    throw new Error(`Port is not supported for health check type: ${props.type}`);
+  }
+}
+
+function ruleEnableSNIIsNotAllowed(props: HealthCheckProps) {
+  if (props.enableSNI) {
+    throw new Error(`EnableSNI is only supported for health check type: ${HealthCheckType.HTTPS}`);
+  }
+}
+
+function ruleChildHealthChecksIsNotAllowed(props: HealthCheckProps) {
+  if (props.childHealthChecks && props.childHealthChecks.length > 0) {
+    throw new Error(`ChildHealthChecks is only supported for health check type: ${HealthCheckType.CALCULATED}`);
+  }
+}
+
+function ruleSearchStringIsNotAllowed(props: HealthCheckProps) {
+  if (props.searchString) {
+    throw new Error(`SearchString is only supported for health check types: ${HealthCheckType.HTTP_STR_MATCH}, ${HealthCheckType.HTTPS_STR_MATCH}`);
+  }
+}
+
+function validateRegions(props: HealthCheckProps) {
+  if (props.regions === undefined) {
+    // if regions are not specified, use the default regions
+    return;
+  }
+
+  if (new Set(props.regions).size < 3) {
+    throw new Error('Regions must be a list of at least three regions');
+  }
+}
+
+function validateFailureThreshold(props: HealthCheckProps) {
+  if (props.failureThreshold !== undefined && (props.failureThreshold < 1 || props.failureThreshold > 10)) {
+    throw new Error('FailureThreshold must be between 1 and 10');
+  }
+}
+
+function validateRequestInterval(props: HealthCheckProps) {
+  if (props.requestInterval !== undefined) {
+    const requestIntervalInSeconds = props.requestInterval.toSeconds();
+    if (requestIntervalInSeconds < 10 || requestIntervalInSeconds > 30) {
+      throw new Error('RequestInterval must be between 10 and 30 seconds');
+    }
+  }
+}
+
+function validateIpAddress(props: HealthCheckProps) {
+  if (props.ipAddress === undefined) {
+    return;
+  }
+
+  if (
+    !new RegExp(
+      '^((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]).){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]))$|^(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]).){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]).){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))$',
+    ).test(props.ipAddress)
+  ) {
+    throw new Error('IpAddress must be a valid IPv4 or IPv6 address');
+  }
```

Comment:
This regex is not human readable IMO. We should leave it out and let CFN to do the validation or please add documentations that link to the CFN page and describe the Regex.